### ISSUE TITLE
Build scripts not available in repository

### ISSUE THREAD

**Issue by TheAssassin** [rand256/valetudo#146]
Combining the steps gained from a few different locations, I can build my own image. That's cumbersome, though, and there's many differences to the released images. For example, I could not yet find out where `cloud-dnsmasq` is coming from.

We should add build scripts to the repository which take an upstream image and perform all necessary patching steps. This way, it's much easier for users to build their own images from the upstream ones, and they can also perform modifications to e.g., the deployment files or add their own pubkey directly into the image.

TL;DR: building an image should be as easy as `./build.sh upstream.pkg`.

**Comment by pidator** [comment]
> For example, I could not yet find out where `cloud-dnsmasq` is coming from.
 
Have a look [@rand256's image builder](https://github.com/rand256/vacuum), should have mentioned this before... ;-)

**Comment by TheAssassin** [comment]
Thanks. We should add a link to the README or the deployment docs.

Edit: I guess the `deployment` directory should be moved to https://github.com/rand256/vacuum, actually. It's kind of "out of date" anyway, for instance the cloud-dnsmasq stuff is added in https://github.com/rand256/vacuum/commit/a1b465efc2a994407cdca247c2c0bb4f70073116#diff-3b44d37488beb62a52b4de10e3d23700 and https://github.com/rand256/vacuum/commit/a1b465efc2a994407cdca247c2c0bb4f70073116#diff-d35071e3eb5bd0cef0df7c9851d5d32c.

**Comment by pidator** [comment]
I think @rand256 was trying to keep this two things separate:
With the valetudo repository (and also the current, outdate deployment section) you can build just the valetudo binary and have the necessary modifications to get valetudo running even if your not touching the rest of the firmware. I was using just these files the last two month on my Gen1 rooted with the originally dustcloud imagebuilder and no other firmware modifications without any problems (This is only possible till a specific firmware release of the robot).

The other repository is just the image building process with the ability to modify the ubuntu system.

In the meantime with xiaomi trying to stop people of rooting the firmware or get to run custom software on the robot it becomes more and more necessary to modify parts of the ubuntu system (iptables, dnsmasq, ...) or miio tools (downgrade to a specific version) so that valetudo could run on the robot at all. For this reason @rand256 or @zvldz are providing espacially these premodified image-releases to get everything work simple. My robot did it's first random-factory-reset after ~2 month with originally image mentioned above. For that reason I've tried the premodified image of rand256 this week the first time.

**Comment by rand256** [comment]
Now images provided in releases section will be built by vanilla [vacuumz builder](https://github.com/zvldz/vacuum), and the old instructions in deployment section are still applicable for building valetudo RE itself.
